build: update toolchain to Fedora 42 with clang 20.1 and libstdc++ 15

avikivity opened the pull request:
Rebase to Fedora 42 with clang 20.1 and libstdc++ 15.

JAVA8_HOME environment variable dropped since we no longer use it.

Disabled weak dependencies in dnf.conf in an attempt to reduce unwanted packages.

cassandra-stress package updated with version that doesn't depend on no-longer-available Java 11.

Since Fedora 42 unified /usr/bin and /usr/sbin, we follow suit
and only install to /usr/bin.

Optimized clang binaries generates and stored in

  https://devpkg.scylladb.com/clang/clang-20.1.3-Fedora-42-aarch64.tar.gz
  https://devpkg.scylladb.com/clang/clang-20.1.3-Fedora-42-x86_64.tar.gz

No backport; we'll keep release branch toolchains stable.

scylladb-promoter commented:
#### :red_circle: CI State: FAILURE

:white_check_mark: - [Framework test](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/17076/)
:x: - [Build](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/17076/)
#### Build Details:
- Duration: 42 min
- Builder: spider4.cloudius-systems.com

avikivity commented:
v2: Since Fedora 42 unified /usr/bin and /usr/sbin, we follow suit and only install to /usr/bin.

scylladb-promoter commented:
#### :red_circle: CI State: FAILURE

:white_check_mark: - [Framework test](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/17078/)
:white_check_mark: - [Build](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/17078/)
:white_check_mark: - [Docker Test](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci-docker/1193/)
:x: - [Rpm Artifact Tests](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci-centos-rpm/488/)
:white_check_mark: - [Deb Artifact Tests](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci-unified-deb/487/)
:x: - [dtest with consistent topology changes](https://jenkins.scylladb.com/job/scylla-master/job/gating-dtest-release-with-consistent-topology-changes/8195/)
:x: - [dtest with gossip topology changes](https://jenkins.scylladb.com/job/scylla-master/job/gating-dtest-release/16263/)
:x: - [Offline-installer Artifact Tests](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci-offline-installer/723/)
:x: - [dtest with tablets](https://jenkins.scylladb.com/job/scylla-master/job/gating-dtest-release-with-tablets/5022/)
:x: - [Unit Tests](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/17078/)
#### Failed Tests (15/43985):
- [test_basic_data_types](https://jenkins.scylladb.com//job/scylla-master/job/scylla-ci/17078/testReport/junit/json_test/TestsFromJsonInsert/Tests___dtest_with_consistent_topology_changes___test_basic_data_types) [:mag:](https://github.com/scylladb/scylladb/issues?q=is:issue+is:open+test_basic_data_types)
- [test_basic_data_types](https://jenkins.scylladb.com//job/scylla-master/job/scylla-ci/17078/testReport/junit/json_test/TestsFromJsonInsert/Tests___dtest_with_gossip_topology_changes___test_basic_data_types) [:mag:](https://github.com/scylladb/scylladb/issues?q=is:issue+is:open+test_basic_data_types)
- [test_basic_data_types](https://jenkins.scylladb.com//job/scylla-master/job/scylla-ci/17078/testReport/junit/json_test/TestsFromJsonInsert/Tests___dtest_with_tablets___test_basic_data_types) [:mag:](https://github.com/scylladb/scylladb/issues?q=is:issue+is:open+test_basic_data_types)
- [test_simple_schema](https://jenkins.scylladb.com//job/scylla-master/job/scylla-ci/17078/testReport/junit/json_test/TestsJsonFullRowInsertSelect/Tests___dtest_with_tablets___test_simple_schema) [:mag:](https://github.com/scylladb/scylladb/issues?q=is:issue+is:open+test_simple_schema)
- [test_simple_schema](https://jenkins.scylladb.com//job/scylla-master/job/scylla-ci/17078/testReport/junit/json_test/TestsJsonFullRowInsertSelect/Tests___dtest_with_consistent_topology_changes___test_simple_schema) [:mag:](https://github.com/scylladb/scylladb/issues?q=is:issue+is:open+test_simple_schema)
- [test_simple_schema](https://jenkins.scylladb.com//job/scylla-master/job/scylla-ci/17078/testReport/junit/json_test/TestsJsonFullRowInsertSelect/Tests___dtest_with_gossip_topology_changes___test_simple_schema) [:mag:](https://github.com/scylladb/scylladb/issues?q=is:issue+is:open+test_simple_schema)
- [test_basic_data_types](https://jenkins.scylladb.com//job/scylla-master/job/scylla-ci/17078/testReport/junit/json_test/TestsToJsonSelect/Tests___dtest_with_tablets___test_basic_data_types) [:mag:](https://github.com/scylladb/scylladb/issues?q=is:issue+is:open+test_basic_data_types)
- [test_basic_data_types](https://jenkins.scylladb.com//job/scylla-master/job/scylla-ci/17078/testReport/junit/json_test/TestsToJsonSelect/Tests___dtest_with_gossip_topology_changes___test_basic_data_types) [:mag:](https://github.com/scylladb/scylladb/issues?q=is:issue+is:open+test_basic_data_types)
- [test_basic_data_types](https://jenkins.scylladb.com//job/scylla-master/job/scylla-ci/17078/testReport/junit/json_test/TestsToJsonSelect/Tests___dtest_with_consistent_topology_changes___test_basic_data_types) [:mag:](https://github.com/scylladb/scylladb/issues?q=is:issue+is:open+test_basic_data_types)
- [test_complex_data_types](https://jenkins.scylladb.com//job/scylla-master/job/scylla-ci/17078/testReport/junit/json_test/TestsToJsonSelect/Tests___dtest_with_tablets___test_complex_data_types) [:mag:](https://github.com/scylladb/scylladb/issues?q=is:issue+is:open+test_complex_data_types)
#### Build Details:
- Duration: 7 hr 13 min
- Builder: spider8.cloudius-systems.com

avikivity commented:
Dependency: https://github.com/scylladb/scylladb/pull/24549

avikivity commented:
- rebased
 - dropped now-unneeded packaging changes
 - dropped upstreamed weak dependencies in docker
 - updated new cassandra-stress version to latest
 - updated clang to 20.1.6

scylladb-promoter commented:
#### :red_circle: CI State: FAILURE

:white_check_mark: - [Framework test](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/17797/)
:x: - [Build](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/17797/)
#### Build Details:
- Duration: 1 hr 3 min
- Builder: spider6.cloudius-systems.com

scylladb-promoter commented:
#### :green_circle: CI State: SUCCESS

#### Build Details:
- Duration: 2 min 37 sec
- Builder: spider6.cloudius-systems.com

scylladb-promoter commented:
#### :green_circle: CI State: SUCCESS

:white_check_mark: - [Framework test](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/17813/)
:white_check_mark: - [Build](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/17813/)
:x: - [Offline-installer Artifact Tests](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci-offline-installer/752/)
:white_check_mark: - [Docker Test](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci-docker/1221/)
:white_check_mark: - [dtest with tablets](https://jenkins.scylladb.com/job/scylla-master/job/gating-dtest-release-with-tablets/5632/)
:x: - [Rpm Artifact Tests](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci-centos-rpm/516/)
:white_check_mark: - [Deb Artifact Tests](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci-unified-deb/515/)
:white_check_mark: - [dtest with consistent topology changes](https://jenkins.scylladb.com/job/scylla-master/job/gating-dtest-release-with-consistent-topology-changes/8817/)
:white_check_mark: - [dtest with gossip topology changes](https://jenkins.scylladb.com/job/scylla-master/job/gating-dtest-release/16900/)
:white_check_mark: - [Unit Tests](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/17813/)
#### Build Details:
- Duration: 8 hr 2 min
- Builder: spider4.cloudius-systems.com

avikivity commented:
Original patchelf commit: https://github.com/NixOS/patchelf/commit/43b75fbc9ffbc1190fee7c8693ad74cb8286cfd4

avikivity commented:
/opt/scylladb/bin/scylla works (likely because we reserved space, 8d777b3ad5a23953fa23988ab0b92ca056d4f9ba), so we only need a solution for binaries we carry over from Fedora unchanged.

avikivity commented:
@michoecho do you see another option? I don't want to revert https://github.com/scylladb/scylladb/commit/4ecce2d2865d70f1b8c20202b600bbd1bbb64e4a (I intend to keep scylla patchelfed in any case).

mykaul commented:
> Optimized clang binaries generates and stored in
> 
> https://devpkg.scylladb.com/clang/clang-20.1.3-Fedora-42-aarch64.tar.gz https://devpkg.scylladb.com/clang/clang-20.1.3-Fedora-42-x86_64.tar.gz

I assume all releng builds use it, since they use dbuild (but worth checking).
Do developers, who may not be using dbuild for whatever reason, are aware of this location?

avikivity commented:
Claude wrote a patchelf substtitue without blinking

avikivity commented:
> > Optimized clang binaries generates and stored in
> > https://devpkg.scylladb.com/clang/clang-20.1.3-Fedora-42-aarch64.tar.gz https://devpkg.scylladb.com/clang/clang-20.1.3-Fedora-42-x86_64.tar.gz
> 
> I assume all releng builds use it, since they use dbuild (but worth checking). Do developers, who may not be using dbuild for whatever reason, are aware of this location?

Those who look at the patch changelog are aware.

nyh commented:
> Those who look at the patch changelog are aware.

Aware of what? I don't understand. Unless I'm missing something, developers who run Scylla locally for tests don't use this patchelf business in the first place. What should be aware of? And which "patch changelog" should they be reading to be aware? :-(

avikivity commented:
> > Those who look at the patch changelog are aware.
> 
> Aware of what? I don't understand. Unless I'm missing something, developers who run Scylla locally for tests don't use this patchelf business in the first place.

@mykaul was talking about the optimized clang. I advertised it on slack long ago, and people who use it outside dbuild have to upgrade.

> What should be aware of? And which "patch changelog" should they be reading to be aware? :-(

"build: update toolchain to Fedora xx with clang yy"

scylladb-promoter commented:
#### :green_circle: CI State: SUCCESS

:white_check_mark: - [Framework test](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/17823/)
:white_check_mark: - [Build](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/17823/)
:white_check_mark: - [Docker Test](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci-docker/1222/)
:white_check_mark: - [Offline-installer Artifact Tests](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci-offline-installer/753/)
:x: - [Deb Artifact Tests](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci-unified-deb/516/)
:x: - [Rpm Artifact Tests](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci-centos-rpm/517/)
:white_check_mark: - [Unit Tests](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/17823/)
#### Build Details:
- Duration: 7 hr 34 min
- Builder: spider6.cloudius-systems.com

michoecho commented:
> Claude wrote a patchelf substtitue without blinking

@avikivity Cool.

> @michoecho do you see another option? I don't want to revert [4ecce2d](https://github.com/scylladb/scylladb/commit/4ecce2d2865d70f1b8c20202b600bbd1bbb64e4a) (I intend to keep scylla patchelfed in any case).

I see you found one.

Note: the `patchelf.cc` only modifies `PT_INTERP`. Unlike `patchelf`, it doesn't correct `.interp`. This is probably fine, but we should be aware that some tool could get confused by it.

(And also, unlike patchelf, it doesn't keep `PT_INTERP`/`.interp` contained within some `PT_LOAD`. But I don't know if anything relies on that).

avikivity commented:
Works from ./install.sh directly, fails from rpm.

Theory:
 - rpm runs some binutil to process iotune (split debuginfo or whatever)
 - one of the deficiencies @michoecho found conflicts with that binutil

avikivity commented:
> > > Works from ./install.sh directly, fails from rpm.
> > > Theory:
> > > ```
> > > * rpm runs some binutil to process iotune (split debuginfo or whatever)
> > > 
> > > * one of the deficiencies @michoecho found conflicts with that binutil
> > > ```
> > 
> > 
> > Hypothesis: when `strip` rewrites the binary, it sets `PT_INTERP` based on `.interp`. If there's no `.interp`, it sets `PT_INTERP` to nothing.
> 
> I just checked, and yes. Actually, `strip` altogether discards the appended file piece containing the interpreter path, since it's not covered by any section.

I'll ask Claude to help.

avikivity commented:
btw, alternative to this stuff: create a new mount namespace, bind-mount the interpreter and all the dependencies where they're expected, exec. But it depends on the kernel supporting those features (and not blocking them inside a container).

michoecho commented:
I just also manually checked (by hex-editing the header) that adjusting `.interp` to match the new `PT_INTERP` is enough to make strip work as expected. You don't need to ensure that `PT_INTERP` is covered by a `PT_LOAD`. Although `strip` will print a benign warning about that:
```
/nix/store/v63bxfiacw082c7ijshf60alvvrpfxsq-binutils-2.44/bin/strip: ../inst/libexec/stk12fiP: warning: allocated section `.interp' not in segment
```

We don't really want to add another `PT_LOAD`, though, because that could require expanding the PHT, and the whole point of this PR is to avoid that.

scylladb-promoter commented:
#### :red_circle: CI State: FAILURE

| Mode | Stage | Status | Node |
|------|-------|--------|------|
|  | [Framework test](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/17932/) | :white_check_mark: | i-081c2b4701021b438 |
| dev | [Build](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/17932/) | :white_check_mark: | spider3.cloudius-systems.com |
|  | [test.py](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/17932/) | :white_check_mark: | spider3.cloudius-systems.com |
| release | [Build](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/17932/) | :white_check_mark: | spider4.cloudius-systems.com |
|  | [test.py](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/17932/) | :white_check_mark: | spider4.cloudius-systems.com |
| debug | [Build](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/17932/) | :white_check_mark: | spider2.cloudius-systems.com |
|  | [test.py](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/17932/) | :white_check_mark: | spider2.cloudius-systems.com |
| dtest | [dtest with tablets](https://jenkins.scylladb.com/job/scylla-master/job/gating-dtest-release-with-tablets/5728/) | :white_check_mark: |  |
|  | [dtest with consistent topology changes](https://jenkins.scylladb.com/job/scylla-master/job/gating-dtest-release-with-consistent-topology-changes/8920/) | :white_check_mark: |  |
|  | [dtest with gossip topology changes](https://jenkins.scylladb.com/job/scylla-master/job/gating-dtest-release/17006/) | :white_check_mark: |  |
| artifacts | [Offline-installer Artifact Tests](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci-offline-installer/755/) | :x: |  |
|  | [Docker Test](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci-docker/1224/) | :white_check_mark: |  |
|  | [Deb Artifact Tests](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci-unified-deb/518/) | :white_check_mark: |  |
|  | [Rpm Artifact Tests](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci-centos-rpm/519/) | :white_check_mark: |  |

#### Build Details:
- Duration: 4 hr 34 min

scylladb-promoter commented:
#### :red_circle: CI State: FAILURE

| Mode | Stage | Status | Node |
|------|-------|--------|------|
|  | [Framework test](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/17945/) | :white_check_mark: | sif |
| dev | [Build](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/17945/) | :white_check_mark: | spider11.cloudius-systems.com |
|  | [test.py](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/17945/) | :x: | spider11.cloudius-systems.com |
| release | [Build](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/17945/) | :white_check_mark: | spider1.cloudius-systems.com |
|  | [test.py](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/17945/) | :white_check_mark: | spider1.cloudius-systems.com |
| debug | [Build](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/17945/) | :white_check_mark: | spider7.cloudius-systems.com |
|  | [test.py](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/17945/) | :white_check_mark: | spider7.cloudius-systems.com |
| dtest | [dtest with gossip topology changes](https://jenkins.scylladb.com/job/scylla-master/job/gating-dtest-release/17019/) | :white_check_mark: |  |
|  | [dtest with tablets](https://jenkins.scylladb.com/job/scylla-master/job/gating-dtest-release-with-tablets/5741/) | :white_check_mark: |  |
|  | [dtest with consistent topology changes](https://jenkins.scylladb.com/job/scylla-master/job/gating-dtest-release-with-consistent-topology-changes/8933/) | :white_check_mark: |  |
| artifacts | [Offline-installer Artifact Tests](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci-offline-installer/756/) | :white_check_mark: |  |
|  | [Deb Artifact Tests](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci-unified-deb/519/) | :white_check_mark: |  |
|  | [Rpm Artifact Tests](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci-centos-rpm/520/) | :white_check_mark: |  |
|  | [Docker Test](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci-docker/1225/) | :white_check_mark: |  |

#### Failed Tests (4/71223):
- [test_kmip_provider.dev.1](https://jenkins.scylladb.com//job/scylla-master/job/scylla-ci/17945/testReport/junit/boost.encryption_at_rest_test/cc/dev___test_py___dev___test_kmip_provider_dev_1) [:mag:](https://github.com/scylladb/scylladb/issues?q=is:issue+is:open+test_kmip_provider.dev.1)
- [test_kmip_provider_broken_config_on_restart.dev.1](https://jenkins.scylladb.com//job/scylla-master/job/scylla-ci/17945/testReport/junit/boost.encryption_at_rest_test/cc/dev___test_py___dev___test_kmip_provider_broken_config_on_restart_dev_1) [:mag:](https://github.com/scylladb/scylladb/issues?q=is:issue+is:open+test_kmip_provider_broken_config_on_restart.dev.1)
- [test_kmip_provider_broken_sstables_on_restart.dev.1](https://jenkins.scylladb.com//job/scylla-master/job/scylla-ci/17945/testReport/junit/boost.encryption_at_rest_test/cc/dev___test_py___dev___test_kmip_provider_broken_sstables_on_restart_dev_1) [:mag:](https://github.com/scylladb/scylladb/issues?q=is:issue+is:open+test_kmip_provider_broken_sstables_on_restart.dev.1)
- [test_kmip_provider_multiple_hosts.dev.1](https://jenkins.scylladb.com//job/scylla-master/job/scylla-ci/17945/testReport/junit/boost.encryption_at_rest_test/cc/dev___test_py___dev___test_kmip_provider_multiple_hosts_dev_1) [:mag:](https://github.com/scylladb/scylladb/issues?q=is:issue+is:open+test_kmip_provider_multiple_hosts.dev.1)

#### Build Details:
- Duration: 4 hr 16 min

avikivity commented:
> #### 🔴 CI State: FAILURE
> Mode	Stage	Status	Node
> [Framework test](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/17945/)	✅	sif
> dev	[Build](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/17945/)	✅	spider11.cloudius-systems.com
> [test.py](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/17945/)	❌	spider11.cloudius-systems.com
> release	[Build](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/17945/)	✅	spider1.cloudius-systems.com
> [test.py](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/17945/)	✅	spider1.cloudius-systems.com
> debug	[Build](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/17945/)	✅	spider7.cloudius-systems.com
> [test.py](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/17945/)	✅	spider7.cloudius-systems.com
> dtest	[dtest with gossip topology changes](https://jenkins.scylladb.com/job/scylla-master/job/gating-dtest-release/17019/)	✅	
> [dtest with tablets](https://jenkins.scylladb.com/job/scylla-master/job/gating-dtest-release-with-tablets/5741/)	✅	
> [dtest with consistent topology changes](https://jenkins.scylladb.com/job/scylla-master/job/gating-dtest-release-with-consistent-topology-changes/8933/)	✅	
> artifacts	[Offline-installer Artifact Tests](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci-offline-installer/756/)	✅	
> [Deb Artifact Tests](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci-unified-deb/519/)	✅	
> [Rpm Artifact Tests](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci-centos-rpm/520/)	✅	
> [Docker Test](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci-docker/1225/)	✅	
> #### Failed Tests (4/71223):
> * [test_kmip_provider.dev.1](https://jenkins.scylladb.com//job/scylla-master/job/scylla-ci/17945/testReport/junit/boost.encryption_at_rest_test/cc/dev___test_py___dev___test_kmip_provider_dev_1) [🔍](https://github.com/scylladb/scylladb/issues?q=is:issue+is:open+test_kmip_provider.dev.1)
> * [test_kmip_provider_broken_config_on_restart.dev.1](https://jenkins.scylladb.com//job/scylla-master/job/scylla-ci/17945/testReport/junit/boost.encryption_at_rest_test/cc/dev___test_py___dev___test_kmip_provider_broken_config_on_restart_dev_1) [🔍](https://github.com/scylladb/scylladb/issues?q=is:issue+is:open+test_kmip_provider_broken_config_on_restart.dev.1)
> * [test_kmip_provider_broken_sstables_on_restart.dev.1](https://jenkins.scylladb.com//job/scylla-master/job/scylla-ci/17945/testReport/junit/boost.encryption_at_rest_test/cc/dev___test_py___dev___test_kmip_provider_broken_sstables_on_restart_dev_1) [🔍](https://github.com/scylladb/scylladb/issues?q=is:issue+is:open+test_kmip_provider_broken_sstables_on_restart.dev.1)
> * [test_kmip_provider_multiple_hosts.dev.1](https://jenkins.scylladb.com//job/scylla-master/job/scylla-ci/17945/testReport/junit/boost.encryption_at_rest_test/cc/dev___test_py___dev___test_kmip_provider_multiple_hosts_dev_1) [🔍](https://github.com/scylladb/scylladb/issues?q=is:issue+is:open+test_kmip_provider_multiple_hosts.dev.1)
> 
> #### Build Details:
> * Duration: 4 hr 16 min

https://github.com/scylladb/scylladb/issues/24574

avikivity commented:
v5:
 - rebased past 07c5edcc30145277e803fec75472faf030b97d97
 - clang 20.1.7
 - updated image name

scylladb-promoter commented:
#### :green_circle: CI State: SUCCESS

| Mode | Stage | Status | Node |
|------|-------|--------|------|
|  | [Framework test](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/18022/) | :white_check_mark: | i-0d4892d006f986843 |
| dev | [Build](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/18022/) | :white_check_mark: | spider1.cloudius-systems.com |
|  | [test.py](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/18022/) | :white_check_mark: | spider1.cloudius-systems.com |
| release | [Build](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/18022/) | :white_check_mark: | spider11.cloudius-systems.com |
|  | [test.py](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/18022/) | :white_check_mark: | spider11.cloudius-systems.com |
| debug | [Build](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/18022/) | :white_check_mark: | spider2.cloudius-systems.com |
|  | [test.py](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/18022/) | :white_check_mark: | spider2.cloudius-systems.com |
| dtest | [dtest with consistent topology changes](https://jenkins.scylladb.com/job/scylla-master/job/gating-dtest-release-with-consistent-topology-changes/8992/) | :white_check_mark: |  |
|  | [dtest with tablets](https://jenkins.scylladb.com/job/scylla-master/job/gating-dtest-release-with-tablets/5798/) | :white_check_mark: |  |
|  | [dtest with gossip topology changes](https://jenkins.scylladb.com/job/scylla-master/job/gating-dtest-release/17076/) | :white_check_mark: |  |
| artifacts | [Docker Test](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci-docker/1228/) | :white_check_mark: |  |
|  | [Offline-installer Artifact Tests](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci-offline-installer/759/) | :white_check_mark: |  |
|  | [Deb Artifact Tests](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci-unified-deb/522/) | :white_check_mark: |  |
|  | [Rpm Artifact Tests](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci-centos-rpm/523/) | :white_check_mark: |  |

#### Build Details:
- Duration: 4 hr 1 min

denesb commented:
@avikivity is this ready to merge? I see all-green CI, including artifacts tests, but I want an explicit confirmation too, given that this is a toolchain update.
